src/@services/BaseService.js

```diff
@@ -0,0 +1,134 @@
+export default class BaseService {
+  static async #call(endpoint, options) {
+    const url = new URL(
+      `${endpoint.startsWith('/') ? '' : '/'}${endpoint}`,
+      import.meta.env.VITE_API_BASE_URL
+    );
+
+    for (let param in options?.query) {
+      const value = options.query[param];
+
+      if (value instanceof Array) {
+        for (let i of value) {
+          if (i?.toString()) {
+            url.searchParams.append(param, i);
+          }
+        }
+      } else if (value?.toString()) {
+        url.searchParams.set(param, value);
+      }
+    }
+
+    const request = {
+      credentials: 'include',
+      headers: {
+        Accept: 'application/json',
+        'accept-language': 'en-us',
+        'content-type': `application/json; charset=utf-8`,
+        ...options?.headers
+      },
+      method: options?.method?.toUpperCase() || 'get'
```

Comment:
```suggestion
      method: options?.method?.toUpperCase() || 'get',
      signal: options?.signal
```

We need it for adding `abortController` to handle multiple search queries fired by the user at the same time